tests/test_gradient.py

```diff
@@ -100,6 +154,7 @@ def test_gradientJ(shape, kernel, space_order):
     :param space_order: order of the spacial discretization scheme
     :return: assertion that the Taylor properties are satisfied
     """
+    clear_cache()
```

Comment:
it's not clear why we need this

Reply:
TRAVIS seems to not clear the cache between test which was making `test_adjointJ` crash. Added it there too as we save the full wavefield for safety

Reply:
I was convinced I had left a comment about this, but I cannot find it

``clear_cache`` should only appear at the end of those tests allocating lots of memory, and should be properly documented.

Even better: here you're calling ``clear_cache`` in two different tests (this and the one above). So I suggest to put these tests within the same class, and add the special ``pytest`` method [setup_method](https://docs.pytest.org/en/2.7.3/xunit_setup.html) with ``clear_cache`` in it (and a suitable comment).

Reply:
at least put it at the end, otherwise the last test execution will leave around some memory

Reply:
Isn't it gonna be ignored as still in the scope of the function?

Reply:
I mean at the of each test, rather than at the beginning